include/module.modulemap

```diff
@@ -0,0 +1,67 @@
+module rcheevos {
+    umbrella header "rcheevos.h"
+    
+    module rapi {
+        module rc_api_editor {
+            header "rc_api_editor.h"
+            export *
+        }
+        
+        module rc_api_info {
+            header "rc_api_info.h"
+            export *
+        }
+        
+        module rc_api_request {
+            header "rc_api_request.h"
+            export *
+        }
+        
+        module rc_api_runtime {
+            header "rc_api_runtime.h"
+            export *
+        }
+        
+        module rc_api_user {
+            header "rc_api_user.h"
+            export *
+        }
+
+        export *
+    }
```

Comment:
I was thinking more along the lines of:
```
module rapi {
    header "rc_api_editor.h"
    header "rc_api_info.h"
    header "rc_api_request.h"
    header "rc_api_runtime.h"
    header "rc_api_user.h"
    export *
}
```

Or doesn't it work that way?

Is there some benefit to keeping them separate? If so, does the new intermediary module provide any benefits?

Reply:
You can put multiple headers into a single module, but the general rule of thumb (at least in Objective-C land) is one module per header.